### PR TITLE
Add Queue<PlatformKind::CUDA>

### PR DESCRIPTION
**Pull request by ye-luo**
~~Please review after merging #4973 and #4974~~

## Proposed changes
Add a new abstraction class Queue to unify code paths.

## What type(s) of changes does this code introduce?
- New feature

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted

### CODE REVIEW

**Comment by ye-luo** [comment]
Test this please

File: src/Platforms/CUDA/CUDALinearAlgebraHandles.h [hunk 1]
```diff
@@ -15,28 +15,27 @@
...
 #include "CUDA/cuBLAS.hpp"
+#include "CUDA/QueueCUDA.hpp"
 
 namespace qmcplusplus
 {
 struct CUDALinearAlgebraHandles : public Resource
 {
   // CUDA specific variables
-  cudaStream_t hstream;
+  compute::Queue<PlatformKind::CUDA> queue;
```
Comment (PDoakORNL) [inline src/Platforms/CUDA/CUDALinearAlgebraHandles.h:25]: feels like now might be the time to change the name of this class. the minute the sycl queue shows up the error messages are all going to be sometimes wrong as well.

**Review by PDoakORNL — CHANGES_REQUESTED** [review]
I think propagating better and coding convention following type names from the previous review through here would be worth it.

File: src/Platforms/CUDA/CUDALinearAlgebraHandles.h (hunk 1 above)
Comment (ye-luo) [reply src/Platforms/CUDA/CUDALinearAlgebraHandles.h:25]: Were you referring to CUDALinearAlgebraHandles? Yes it has to be changed. It is not clear how exactly. I probably will demolish this class. So far it is still CUDA only and should not be renamed in this PR.

Comment (PDoakORNL) [reply src/Platforms/CUDA/CUDALinearAlgebraHandles.h:25]: Yes I was referring to this.

**Review by prckent — COMMENTED** [review]
(Small initial comment) I prefer that we don't call things Queues if they are not STL Queues. Perhaps an additional prefix/suffix or different name?

**Comment by ye-luo** [comment]
> (Small initial comment) I prefer that we don't call things Queues if they are not STL Queues. Perhaps an additional prefix/suffix or different name?

I added namespace protection for this purpose. It is `qmcpusplus::compute::Queue` in full

**Comment by PDoakORNL** [comment]
I think SYCL calls them queues but I'm not sure the semantics are really the same as std::queue.  MAGMA also calls their streams queues but there I think the semantics are closer to std::queue.

**Comment by ye-luo** [comment]
@PDoakORNL rebased and reduced code change. Ready for final review.

**Review by PDoakORNL — APPROVED** [review]

**Comment by PDoakORNL** [comment]
This looks good to me now. I do think we should try to get encourage use of the `compute::` in future code that uses this.

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**